### PR TITLE
r.in.ahn: new addon to import the AHN DTM or DSM

### PR DESCRIPTION
**Pull request by ecodiv**
Addon to import the 'Actueel Hoogtebestand Nederland' in GRASS GIS. The AHN consists of high resolution (0.5m) DTM and DSM layers for the Netherlands.  The addon uses r.in.wcs in the background, mostly taking care of setting the region to have the right resolution and extent, so it is aligned with the AHN data (thus behaving like if you would import the raster data with r.in.gdal).

### CODE REVIEW

File: src/raster/r.in.ahn/r.in.ahn.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    try:
+        gs.find_program("r.in.wcs")
+    except Exception as e:
+        gs.fatal(_("The addon r.in.wcs is not installed. Exit"))
+
+    # Get coordinates current region extent
+    if flags["g"]:
+        region_original = create_temporary_name("region_")
+        gs.run_command("g.region", save=region_original)
+    region_current = gs.parse_command("g.region", flags="gu")
```
Comment (petrasovaa) [inline src/raster/r.in.ahn/r.in.ahn.py]: You should probably use `gs.use_temp_region` and `gs.del_temp_region` in cleanup.

File: src/raster/r.in.ahn/r.in.ahn.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            if found["file"]:
+                gs.run_command(
+                    "g.remove", flags="f", type=element, name=map_name, quiet=True
+                )
+
+
+def create_temporary_name(prefix):
+    tmpf = f"{prefix}{str(uuid.uuid4().hex)}"
+    CLEAN_LAY.append(tmpf)
+    return tmpf
```
Comment (petrasovaa) [inline src/raster/r.in.ahn/r.in.ahn.py]: There is a somewhat better way to deal with temporary maps, see the [section Temporary Maps in this notebook](https://github.com/wenzeslaus/foss4g-2022-developing-custom-grass-tools/blob/main/notebook_3_best_practices.ipynb) (eventually will be part of official doc...)

File: src/raster/r.in.ahn/r.in.ahn.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+# % type: string
+# % label: product
+# % description: Choose which product to download (dsm or dtm)
+# % options: dsm, dtm
+# % required: yes
+# %end
+
+# %option G_OPT_R_OUTPUT
+# % key_desc: File name
+# % label: Name of output raster layer
```
Comment (petrasovaa) [inline src/raster/r.in.ahn/r.in.ahn.py]:
```suggestion
```

File: src/raster/r.in.ahn/r.in.ahn.py (hunk 1 above)
Comment (ecodiv) [reply src/raster/r.in.ahn/r.in.ahn.py]: done

File: src/raster/r.in.ahn/r.in.ahn.py (hunk 2 above)
Comment (ecodiv) [reply src/raster/r.in.ahn/r.in.ahn.py]: used the append_node_pid function as recommended.

**Comment by ecodiv** [comment]
@petrasovaa, thanks for the suggestions

**Pull request merged**